compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -1724,10 +1724,10 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
         let mut variant_resolution = None;
         if let ty::Adt(adt_def, _) = qself_ty.kind() {
             if adt_def.is_enum() {
-                let variant_def = adt_def
-                    .variants
-                    .iter()
-                    .find(|vd| tcx.hygienic_eq(assoc_ident, vd.ident, adt_def.did));
+                let variant_def = adt_def.variants.iter().find(|vd| {
+                    let variant_ident = Ident::new(vd.name, tcx.def_ident_span(vd.def_id).unwrap());
+                    tcx.hygienic_eq(assoc_ident, variant_ident, adt_def.did)
```

Comment:
```suggestion
                    tcx.hygienic_eq(assoc_ident, vd.ident(tcx), adt_def.did)
```
?

Reply:
Thanks - I wrote this before I added the `ident` method.

Reply:
Likewise, should `vd.ident(self)` be lifted out of the loop?